simulation/amaru-sim/src/simulator/simulate.rs

```diff
@@ -265,12 +270,38 @@ pub fn simulate<Msg, F>(
         Ok(())
     });
     match result {
-        Ok(_) => (),
+        Ok(_) => {
+            let now = SystemTime::now();
+            let mut file = File::create(format!(
+                "success-{}.trace",
+                now.duration_since(SystemTime::UNIX_EPOCH)
+                    .unwrap()
+                    .as_secs()
+            ))
+            .expect("Failed to create temporary file");
+            for bytes in trace_buffer.lock().iter() {
+                file.write_all(bytes)
+                    .expect("Failed to write JSON data to file");
+            }
```

Comment:
Since we want the success status to be determined by the test and not by possible file writing problems, I’d put this code into a separate function, pass in the name prefix, return a result and then print any errors to stderr. Then this function can also be used in the error case and file writing errors won’t prevent printing the actual test failure.